CMakeLists.txt

```diff
@@ -26,6 +26,7 @@ function(set_default_warning_settings target)
         if(CMAKE_CXX_COMPILER_ID MATCHES "Clang") # match both "Clang" and "AppleClang"
             target_compile_options(${target} PRIVATE
                 -Wc++17-compat-pedantic
+                -Wc++20-compat-pedantic
```

Comment:
fyi I just checked if there's already the same for C++23 (which apparently got finished July this year), but doesn't seem to be there yet https://clang.llvm.org/docs/DiagnosticsReference.html

Reply:
oh! But there's this https://clang.llvm.org/docs/DiagnosticsReference.html#wpre-c-23-compat-pedantic
Maybe we should put that in there for good measure. I think clang is fine with stuff it doesn't know (unlike GCC)